src/ui/components/NetworkMonitor/Filter.tsx

```diff
@@ -0,0 +1,73 @@
+import classNames from "classnames";
+import { useState } from "react";
+
+import MaterialIcon from "../shared/MaterialIcon";
+import Icon from "../shared/Icon";
+
+import { CanonicalRequestType, RequestTypeOptions } from "./utils";
+
+export function Filter({
```

Comment:
If we were to go with this approach, it'd be nice if this were called something else, and have a separate `Filter` (or `FilterDrawer`?) component.